server/autolink/lib_jira_test.go

```diff
@@ -154,6 +154,52 @@ var jiraTests = []linkTest{
 		"http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5",
 		"http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5",
 	},
+	{
+		"Jump To Comment With Jira Cloud Failed",
+		autolink.Autolink{
+			Pattern:  "(https://mmtest.atlassian.net/browse/)(DP)(-)(?P<jira_id>\\d+)[?](focusedCommentId)(=)(?P<comment_id>\\d+)",
+			Template: "[DP-${jira_id} With Focused Comment($comment_id)](https://mmtest.atlassian.net/browse/DP-${jira_id}",
+		},
+		"https://mmtest.atlassian.net/browse/DP-454",
+		"https://mmtest.atlassian.net/browse/DP-454",
+	},
+	// Trials Linker
+	{
+		"Trial With Jira Cloud",
+		autolink.Autolink{
+			Pattern:  "(?P<URI>.*/)(?P<IssueKey>[A-Za-z]+-[0-9]+)[?](focusedCommentId=*(?P<FocusedCommentId>[^\\s&]+))",
+			Template: "[${IssueKey} FocusedComment(${FocusedCommentId})](${URI}${IssueKey}?focusedCommentId=${FocusedCommentId})",
+		},
+		"https://mmtest.atlassian.net/browse/DP-454?focusedCommentId=11347",
+		"[DP-454 FocusedComment(11347)](https://mmtest.atlassian.net/browse/DP-454?focusedCommentId=11347)",
+	},
+	{
+		"Trial With Jira Server #1",
+		autolink.Autolink{
+			Pattern:  "(?P<URI>.*/)(?P<IssueKey>[A-Za-z]+-[0-9]+)[?](focusedCommentId=*(?P<FocusedCommentId>[^\\s&]+))",
+			Template: "[${IssueKey} FocusedComment(${FocusedCommentId})](${URI}${IssueKey}?focusedCommentId=${FocusedCommentId})",
+		},
+		"http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5?focusedCommentId=10200",
+		"[DKHPROJ-5 FocusedComment(10200)](http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5?focusedCommentId=10200)",
+	},
+	{
+		"Trial With Jira Server #2", //Not Worked here ?
+		autolink.Autolink{
+			Pattern:  "(?P<URI>.*/)(?P<IssueKey>[A-Za-z]+-[0-9]+)[?](focusedCommentId=*(?P<FocusedCommentId>[^\\s&]+))",
+			Template: "[${IssueKey} FocusedComment(${FocusedCommentId})](${URI}${IssueKey}?focusedCommentId=${FocusedCommentId})",
+		},
+		"http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5?focusedCommentId=10200&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-10200",
+		"http://ec2-54-157-116-101.compute-1.amazonaws.com/browse/DKHPROJ-5?focusedCommentId=10200&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-10200",
+	},
+	{
+		"Trial With Jira Cloud Without Focused Comment ID", //Not Worked here ?
+		autolink.Autolink{
+			Pattern:  "(?P<URI>.*/)(?P<IssueKey>[A-Za-z]+-[0-9]+)[?](focusedCommentId=*(?P<FocusedCommentId>[^\\s&]+))",
+			Template: "[${IssueKey} FocusedComment(${FocusedCommentId})](${URI}${IssueKey}?focusedCommentId=${FocusedCommentId})",
+		},
+		"https://mmtest.atlassian.net/browse/DP-454",
+		"https://mmtest.atlassian.net/browse/DP-454",
+	},
```

Comment:
hi @mickmister did you mean it will be like this ?

Reply:
Since we want to continue to be able to transform regular issue URLs like this one above, we'll need to incorporate the existing regexes used in the Jira plugin repo [here](https://github.com/mattermost/mattermost-plugin-jira/blob/4db5740c99a287423f72279d46604c0d8d30c0b2/server/plugin.go#L359-L373) in these tests. One of the existing regexes defined in the Jira plugin is able to parse this URL in the test above.

Since we have 3 regexes now, we need to have 3x3 = 9 unit tests for the following:

- The first regex correctly transforms a link its meant to transform
- The first regex does not change a link that the second regex is meant to transform
- The first regex does not change a link that the third regex is meant to transform

and so on for the second and third regexes, for a total of 9 tests.

Also, the regexes in the Jira plugin use the Jira instance's URL in the `Pattern`, so you'll want to do that here as well, instead of using `?P<URI>.*/`.